research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,418 @@
+[node document](https://dom.spec.whatwg.org/#concept-node-document) as the `popup` element.
+
+**Example:**
+
+```html
+<button id="myButton">Anchor element</button>
+<popup open anchor="myButton">
+  <p>
+    <strong>New!</strong>
+    I’m some sort of educational UI…
+  </p>
+</popup>
+```
+
+The target of the `anchor` attribute participates in determining whether a candiate _popup_ element
+is "nested" in a currently-shown `popup`; refer to <a href="#hiding-current-popups">hiding
+currently-shown <code>popup</code> elements steps</a>.
+
+<p class="todo">
+  Whenever there is a spec to point to for CSS anchored positioning, call out here that the anchor
+  attribute is used to help resolve CSS <code>anchor()</code> functions.
+</p>
+
+## Hiding a `popup` element
+
+### The `hide()` method
+
+The `hide()` method hides a `popup` element. When `hide()` is invoked, the user agent must run
+[hiding a `popup` element steps](#hiding-popup-steps) from the `popup` element _subject_ on which
+the method was invoked.
+
+### Light dismissal
+
+"Light dismiss" describes the behavior of a component which dismisses itself automatically when a
+user performs an action that implies that they are finished interacting with the component. The
+actions that trigger light dismiss are:
+
+- The user presses the escape key.
+- A focus change occurs (because of either user interaction or script), where the focus target is
+  outside of the contents of the component. This includes the case where the user invokes a
+  non-focusable element, which causes focus to jump to the nearest focusable ancestor of that element.
+- An OS-level focus change occurs such that the window containing the component no longer has focus.
+  For example, the user switches to a different browser tab or switches to a different application.
+
+<p class="todo">
+  Land <a href="https://github.com/openui/open-ui/pull/243">#243</a> and link out to this canonical
+  definition of light dismissal for consistency (remove inline definition). Layout changes to the
+  popup or its anchor element are also ground for light dismissal, figure out how to surface this
+  special case.
+</p>
+
+When a `popup` element is shown and a light dismiss interaction occurs:
+
+1. If a keypress event for the `ESCAPE` key was fired, the user agent must set the _subject_ to
+   the `popup` element and run [hiding a `popup` element steps](#hiding-popup-steps).
+2. If a focus change occurs, the user agent must set the _start node_ to the element which received
+   the focus event and run [Hiding currently-shown `popup` elements steps](#hiding-current-popups).
+3. Otherwise, the user agent must set the _start node_ to null and run [Hiding currently-shown
+   `popup` elements steps](#hiding-current-popups).
+
+<h3 id="hiding-current-popups">
+  Hiding currently-shown <code>popup</code> elements steps
+</h3>
+
+Starting at the top of the [popup stack](#popup-stack):
```

Comment:
It's not clear what this means.

Reply:
Can you help me understand what's unclear?

Reply:
Well, first, do stacks grow up or down? :) It's not clear which is the top.

Second, which popup stack? This algorithm probably needs to take a document as an input, so you can look at that document's popup stack in particular.

Reply:
Cool, added a reference to the popup's node document in ccd9b75036cb67e177ad11b2bf72b0b9c1ed0afd, but I'll see what else I can do in the popup-stack section to clarify.